bin/varnishd/http2/cache_http2_proto.c

```diff
@@ -743,7 +753,23 @@ h2_rx_headers(struct worker *wrk, struct h2_sess *h2, struct h2_req *r2)
 	}
 	CHECK_OBJ_NOTNULL(r2, H2_REQ_MAGIC);
 
-	if (r2->state != H2_S_IDLE)
+	if (r2->state == H2_S_OPEN) {
+		AN(r2->req->req_body_status);
```

Comment:
This assertion is confusing. Consider making it `assert(r2->req->req_body_status != BS_INIT)` instead.